apis/v1beta1/shared_types.go

```diff
@@ -243,6 +243,40 @@ type PortNumber int32
 // ReferenceGrant object is required in the referent namespace to allow that
 // namespace's owner to accept the reference. See the ReferenceGrant
 // documentation for details.
+//
+// <gateway:experimental:description>
+//
+// When the BackendRef points to a Kubernetes Service implementations MUST allow
+// the backend protocol to be specified by:
+//   - setting the protocol field on the Service's ServicePort
+//   - setting the protocol field on the Service's related Endpoint/EndpointSlice's EndpointPort
+//   - setting the appProtocol field on the Service's ServicePort
+//   - setting the appProtocol field on the Service's related Endpoint/EndpointSlice's EndpointPort
+//
+// For appProtocol implementations MUST recognize the Kubernetes Standard Application Protocols
+// defined in [KEP-3726]. This supports IANA standard service names and extra constants defined
+// in the KEP that have a prefix of "kubernetes.io/". Gateway API MAY define additional
+// constants with the prefix "gateway.networking.k8s.io/"
+//
+// If a Service appProtocol isn't specified an implementation MAY infer the backend
+// protocol through its own means. Implementations MAY infer the protocol from the
+// Route type referring to the backend Service.
+//
+// Implementations MAY support multiplexing TCP and UDP on the same port. Otherwise
+// implementations MUST set ResolvedRefs condition to False with the Reason UnsupportedProtocol
+// with a clear message that multiplexing is not supported.
+//
+// If a Route attached to a Gateway is not able to send traffic to the backend using
```

Comment:
Why is this unique to a Gateway? Is the expected behavior different for mesh? (I don't actually know what the answer here should be)

/cc @howardjohn @keithmattix

Reply:
Honestly forgot about mesh - I can drop the Gateway ref in this sentence

Reply:
Updated the language - I would be suprised if we treated mesh and non-mesh differently here